OptimizelySDK/Notifications/NotificationCenter.cs

```diff
@@ -0,0 +1,294 @@
+        /// <param name="variation">The variation entity</param>
+        public delegate void FeatureExperimentCallback(string featureKey, string userId, UserAttributes userAttributes,
+            Experiment experiment, Variation variation);
+
+        /// <summary>
+        /// Delegate for feature rollout notifcations.
+        /// </summary>
+        /// <param name="featureKey">The feature key</param>
+        /// <param name="userId">The user identifier</param>
+        /// <param name="userAttributes">Associative array of attributes for the user</param>
+        /// <param name="audiences">Array of audience</param>
+        public delegate void FeatureRolloutCallback(string featureKey, string userId, UserAttributes userAttributes,
+            Audience[] audiences);
+
+        private ILogger Logger;
+
+        // Notification Id represeting number of notifications.
+        public int NotificationId { get; private set; } = 1;
+
+        // Associative array of notification type to notification id and notification pair.
+        private Dictionary<NotificationType, Dictionary<int, object>> Notifications = 
+            new Dictionary<NotificationType, Dictionary<int, object>>();
+
+        /// <summary>
+        /// Property representing total notifications count.
+        /// </summary>
+        public int NotificationsCount
+        {
+            get
+            {
+                int notificationsCount = 0;
+                foreach (var notificationsMap in Notifications.Values)
+                {
+                    notificationsCount += notificationsMap.Count;
+                }
+
+                return notificationsCount;
+            }
+        }
+
+        /// <summary>
+        /// NotificationCenter constructor
+        /// </summary>
+        /// <param name="logger">The logger object</param>
+        public NotificationCenter(ILogger logger = null)
+        {
+            Logger = logger ?? new NoOpLogger();
+
+            foreach (NotificationType notificationType in Enum.GetValues(typeof(NotificationType)))
+            {
+                Notifications[notificationType] = new Dictionary<int, object>();
+            }
+        }
+
+        /// <summary>
+        /// Add a notification callback of decision type to the notification center.
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="decisionCallBack">Callback function to call when event gets triggered</param>
+        /// <returns>int | 0 for invalid notification type, -1 for adding existing notification
+        /// or the notification id of newly added notification.</returns>
+        public int AddNotification(NotificationType notificationType, ActivateCallback activateCallback)
+        {
+            if (!IsNotificationTypeValid(notificationType, NotificationType.Activate))
+                return 0;
+
+            return AddNotification(notificationType, (object)activateCallback);
+        }
```

Comment:
I may remove overloaded AddNotification methods but it is typed safe that's why I added it. Otherwise AddNotification(NotificationType, Delegate) can be used as an argument type to support all subtype delegates.
Using Microsoft C# events, approach will be different for Add/Remove and no notificationId & notificationtype concept would be there. Here approach is same with other SDKs.